collector/hwmon_linux.go

```diff
@@ -18,19 +18,32 @@ package collector
 
 import (
 	"errors"
+	"fmt"
 	"os"
 	"path/filepath"
 	"regexp"
 	"strconv"
 	"strings"
 
+	"github.com/alecthomas/kingpin/v2"
 	"github.com/go-kit/log"
 	"github.com/go-kit/log/level"
 	"github.com/prometheus/client_golang/prometheus"
 	"golang.org/x/sys/unix"
 )
 
 var (
+	hwmonIncludeSet bool
+	hwmonInclude    = kingpin.Flag("collector.hwmon.unit-include", "Regexp of hwmon units to include. Units must both match include and not match exclude to be included.").Default(".+").PreAction(func(c *kingpin.ParseContext) error {
```

Comment:
This should probably be `chip-name` not `unit`. This is based on how we map `/sys/class/hwmon/*/name` to metrics like `node_hwmon_chip_names{chip="platform_coretemp_0",chip_name="coretemp"}`.

```suggestion
	hwmonInclude    = kingpin.Flag("collector.hwmon.chip-name-include", "Regexp of hwmon names to include. Names must both match include and not match exclude to be included.").Default(".+").PreAction(func(c *kingpin.ParseContext) error {
```